mllib/src/main/scala/org/apache/spark/ml/tuning/ValidatorParams.scala

```diff
@@ -67,6 +71,39 @@ private[ml] trait ValidatorParams extends HasSeed with Params {
   /** @group getParam */
   def getEvaluator: Evaluator = $(evaluator)
 
+  /**
+   * param to control the number of models evaluated in parallel
+   * Default: 1
+   *
+   * @group param
+   */
+  val numParallelEval: IntParam = new IntParam(this, "numParallelEval",
+    "max number of models to evaluate in parallel, 1 for serial evaluation",
+    ParamValidators.gtEq(1))
+
+  /** @group getParam */
+  def getNumParallelEval: Int = $(numParallelEval)
+
+  /**
+   * Creates a execution service to be used for validation, defaults to a thread-pool with
+   * size of `numParallelEval`
+   */
+  protected var executorServiceFactory: (Int) => ExecutorService = {
+    (requestedMaxThreads: Int) => ThreadUtils.newDaemonCachedThreadPool(
```

Comment:
So my thinking was that if the thread calling fit is terminated, it would have to be the JVM shutting down which would exit without waiting for these daemon threads.  We don't really at what point the daemon threads stop or if they stop abruptly since any unfinished work is useless.  So I'm not sure if adding a shutdownHook would do anything different?

On the other hand, if the SparkSession wanted to cancel the running threads with the JVM still running, I think it could do that if it provided it's own ExecutorService.